### DISCUSSION TITLE
Update browser URL without navigating? (use-case: update after auto-save in editor)

### DISCUSSION THREAD

**Discussion by wondering639** [remix-run/remix#10378] [Q&A] [answered]
In my app I have an text editor which periodically automatically saves. On saving I want to update the URL in the beowser address bar (without causing a navigation, because the user shall still be able to edit after auto saving!), so the user can copy&paste the URL from the browser address bar to get back to the saved text, share it etc.
I did a test with `window.history.pushState` that worked fine - it updates the address without causing navigation. I just want to know if that is the correct way to do it or if there is a more "remix like" way (any hook maybe?). Thanks!

**Comment by gustavopch** [comment]
If `navigate('.', { replace: true, preventScrollReset: true })` doesn't do what you want, then `window.history.pushState` is a good option. Remix/RR's philosophy leverages native APIs as much as possible, it doesn't intend to wrap everything or have its own way to do everything.

**Reply by wondering639** [reply]
Thanks for your reply. While `navigate('.', { replace: true, preventScrollReset: true })` does not reload the page, it does not add an id like `/12345`. If I use `navigate('/12345', { replace: true, preventScrollReset: true })` it of course does a real navigation. Or am I missunderstanding how it should be use?

**Reply by gustavopch** [reply]
What's the problem that the navigation causes that you want to avoid?

**Reply by wondering639** [reply]
> What's the problem that the navigation causes that you want to avoid?

I need to update the URL without actually navigating. Use-case: editor with automatic save. User creates a new document and after first automatic save, URl shall be updated to include the id of the saved document (while user continues to work on the document). `window.history.pushState` seems to work fine, just wanted to double check I did not miss any "native" remix functionality :-)

**Reply by gustavopch** [reply, marked as answer]
Yeah, `window.history.pushState` is fine then.
